pkg/apis/core/v1alpha1/types_shoot.go

```diff
@@ -1239,9 +1239,11 @@ type Maintenance struct {
 // MaintenanceAutoUpdate contains information about which constraints should be automatically updated.
 type MaintenanceAutoUpdate struct {
 	// KubernetesVersion indicates whether the patch Kubernetes version may be automatically updated (default: true).
-	KubernetesVersion bool `json:"kubernetesVersion" protobuf:"varint,1,opt,name=kubernetesVersion"`
+	// +optional
+	KubernetesVersion *bool `json:"kubernetesVersion,omitempty" protobuf:"varint,1,opt,name=kubernetesVersion"`
```

Comment:
For the Kubernetes version, it could remain as is, right? No need to make it optional.

Reply:
Yeah, but just for consistency reasons.

Reply:
I don't understand this argument in this context. Why is consistency relevant? It's actually wrong to make this optional since it will be needed in all cases (workerless or not), just like before.

Reply:
Okay, will revert it.